rs/tests/driver/src/driver/vector_vm.rs

```diff
@@ -351,3 +366,81 @@ impl Serialize for VectorTransform {
         s.end()
     }
 }
+
+fn add_custom_target(
+    sources: &mut BTreeMap<String, VectorSource>,
+    transforms: &mut BTreeMap<String, VectorTransform>,
+    target_id: String,
+    ip: IpAddr,
+    labels: Option<BTreeMap<String, String>>,
+) {
+    let source = VectorSource::new(target_id.clone(), ip);
+    let source_key = format!("{}-source", target_id);
+
+    let mut extended_labels = labels.unwrap_or_default();
+    extended_labels.extend([
+        (IC_NODE.to_string(), target_id.clone()),
+        (ADDRESS.to_string(), ip.to_string()),
+    ]);
+
+    let transform = VectorTransform::new(source_key.clone(), extended_labels);
+
+    sources.insert(source_key, source);
+    transforms.insert(format!("{}-transform", target_id), transform);
+}
+
+pub trait HasVectorTargets {
+    // Adds a custom vector target for scraping logs.
+    // The target has to have systemd-journal-gatewayd open on port 19531.
+    fn add_custom_vector_target(
+        &self,
+        target_id: String,
+        ip: IpAddr,
+        labels: Option<BTreeMap<String, String>>,
+    ) -> anyhow::Result<()>;
+}
+
+impl HasVectorTargets for TestEnv {
+    fn add_custom_vector_target(
+        &self,
+        target_id: String,
+        ip: IpAddr,
+        labels: Option<BTreeMap<String, String>>,
+    ) -> anyhow::Result<()> {
+        // Get current targets, if any
+        let vector_dir = self.get_path("vector");
+        std::fs::create_dir_all(&vector_dir).map_err(anyhow::Error::from)?;
+        let current_custom_targets = vector_dir.join("custom_targets.json");
+
+        let mut custom_targets = self.get_custom_vector_targets()?;
+
+        custom_targets.insert(target_id, CustomTarget { ip, labels });
+
+        std::fs::write(
+            current_custom_targets,
+            serde_json::to_string_pretty(&custom_targets)?,
+        )
+        .map_err(anyhow::Error::from)
```

Comment:
You could consider adding a dedicated struct for `CustomVectorTargets` and implementing [TestEnvAttribute](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/tests/driver/src/driver/test_env.rs?L192) for it. That will then take care of writing and reading it from the `TestEnv`.